pkg/minikube/cluster/cluster.go

```diff
@@ -340,6 +339,9 @@ func createHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
 		return nil, fmt.Errorf("Error attempting to save store: %s", err)
 	}
 
+	// Create instance-data.json since machine path is available
+	instancedata.Data.Create()
```

Comment:
-1, I still don't get why this should be needed here. This seems absolutely random to me.